src/lib/utilities/request-from-api.ts

```diff
@@ -77,7 +83,11 @@ export const requestFromAPI = async <T>(
       if (onError && isFunction(onError)) {
         onError({ status, statusText, body });
       } else {
-        throw new Error(`${status}: ${statusText}`);
+        throw {
+          statusCode: response.status,
+          statusText: response.statusText,
+          response,
+        } as NetworkError;
```

Comment:
I think you can't catch a type, so I don't know if the `as` does anything. But, I could be wrong and this isn't a hill that I'm interested in dying on.

Reply:
The type of this won't end up elsewhere, but it's here to type check the object we're throwing so someone in the future will get an error if they add a "NewKey" without updating the type interface.